backend/middleware/authorisationMiddleware.ts

```diff
@@ -0,0 +1,25 @@
+import jwtDecode from 'jwt-decode'
+import type { RequestHandler } from 'express'
+import logger from '../log'
+
+export default function authorisationMiddleware(authorisedRoles: string[] = []): RequestHandler {
+  return (req, res, next) => {
+    if (res.locals?.access_token) {
+      const { authorities: roles = [] } = jwtDecode(res.locals.access_token) as { authorities?: string[] }
+
+      if (authorisedRoles.length && !roles.some((role) => authorisedRoles.includes(role))) {
```

Comment:
no authorisedRoles are passed in at the moment - as this file grows we can possibly specify the only ones allowed